main.go

```diff
@@ -232,6 +232,15 @@ func (h k8sHelpers) GetNode(cli *client.Client) (*api.Node, error) {
 }
 
 func (h k8sHelpers) AddLabels(n *api.Node, labels Labels) {
+	// Remove nfd node labels that no longer apply
```

Comment:
If you move the part that removes labels I don't think you would want to rename `AddLabels` as it's behavior hasn't changed.

Does it make sense to split this into another function? We'd be changing the set of calls to update labels to:
```go
helper.RemoveNFDLabels(node)
helper.AddLabels(node, labels)
```
instead of just doing:
```go
helper.UpdateNodeLabels(node, labels)
```

You can definitely do it that way but I'm not sure what value you gain by splitting it up since the behavior will always be done together.